react-redux/src/main/scala/demo/ReduxFacade.scala

```diff
@@ -27,6 +28,13 @@ object ReduxFacade {
     protected val _type: String
     def unapply(a: Action[String]): Option[T] =
       if (a.`type` == _type) Some(a.asInstanceOf[T]) else None
+
+    @scala.inline
```

Comment:
this will enable you to make invalid objects for any subclass which has more than the `type` member, so it's not a good pattern